### ISSUE TITLE
Extend tap-runner plugin configuration to allow using "any" testing framework with TAP support

### ISSUE THREAD

**Issue by ericcornelissen** [stryker-mutator/stryker-js#4266]
**Is your feature request related to a problem? Please describe.**
Currently the [`tap-runner`](https://stryker-mutator.io/docs/stryker-js/tap-runner/) is limited to test frameworks that operate in a separate way (notably [`node-tap`](https://node-tap.org/) and the [Node.js built-in test runner](https://nodejs.org/api/test.html)). Other test frameworks (at least [AVA](https://github.com/avajs/ava)) that support the Test Anything Protocol (TAP) may not work due to the [way the plugin spawns process](https://github.com/stryker-mutator/stryker-js/blob/b159f73a9b8327c6cc1694a4c1b76b58d7701430/packages/tap-runner/src/tap-test-runner.ts#L128). Extending the `tap-runner` plugin to support more testing frameworks that support the TAP could provide a relatively lightweight way to have _some_ integration with more frameworks.

**Describe the solution you'd like**
I see two options:

1. Add an option to the `tap-runner` plugin to select the testing framework and implement a runner for each supported framework. The configuration could look something like:

   ```js
   { 
     testRunner: "tap",
     tap: {
       testFiles: [/* bla */],
       nodeArgs: [/* bla */],
       framework: "ava", // or "node-tap" or "node-native", probably defaults to "node-native"
     },
   }
   ```

   The main benefit of this I see is that it's simpler to understand, especially for users. The main drawback is that every framework has to be supported by manually adding a runner for it.

2. Extend the `tap-runner` plugin configuration to allow:
   - Specifying a "program" to run and corresponding "programArgs".
   - Specifying an alternative way to inject the `TapTestRunner.hookFile`.
   
   This could look something like:

   ```js
   { 
     testRunner: "tap",
     tap: {
       program: "./node_modules/.bin/ava",
       programArgs: ["--tap", "--node-arguments='-r %HOOKFILE%'"]
     },
   }
   ```

   If we go this route, I'd additionally suggest including `./node_modules/.bin` in the `PATH` so invoking any available commands is more convenient.

**Describe alternatives you've considered**
I described two possibilities above. A dedicated runner for AVA would also work for me, this has been discussed before in issues and discussions. I think supporting AVA and other frameworks with TAP support in this way would be beneficial regardless.

**Additional context**
See the discussion over at #3245 for more details and context.

**Comment by nicojs** [comment]
I've discussed this with @danny12321 . I think we'll go with this setup:

```json
  "tap": {
    "hardBail": false,
    "nodeArgs": ["./node_modules/.bin/ava", "--tap", "--node-arguments='-r {{hooksFile}}'", "{{testFile}}"]
  },
```

So with a bit of templating magic: 

| var | replace | default |
| -|-|-|
| `'{{hooksFile}}'` | The actual location of the current hookfile | By default it will be prepended `['-r', '{{hooksFile}}', ...]`|
| `'{{testFile}}'` | The test file to run | By default it will appended last: `[..., '{{testFile}}']` |

**Comment by nicojs** [comment]
Closed with #4283
